### PR TITLE
Corrected retrieval of store params

### PR DESCRIPTION
**Pull request by TonioF**
Follow-up PR to #535 and #600 . It addresses the problem that a dataset configured in a config's dataset section to use an s3 file system would cause an error.

### CODE REVIEW

**Review by AliceBalfanz — APPROVED** [review]
Works nice! Thanks for fixing :)

File: xcube/webapi/context.py [hunk 1]
```diff
@@ -352,28 +354,33 @@ def _maybe_assign_store_instance_ids(self):
...
     def _get_other_store_params_than_root(self,
                                           dataset_config: DatasetConfigDict) \
             -> Dict:
         if FS_TYPE_TO_PROTOCOL.get(dataset_config.get('FileSystem',
                                                       'file')) != 's3':
             return {}
-        store_params = dict()
+        store_params = dict(
+            storage_options=dict()
+        )
```
Comment (forman) [inline xcube/webapi/context.py:374]:
```suggestion
        storage_options=dict()
```

File: xcube/webapi/context.py [hunk 2]
```diff
@@ -352,28 +354,33 @@ def _maybe_assign_store_instance_ids(self):
...
         if 'Anonymous' in dataset_config:
-            store_params['anon'] = dataset_config['Anonymous']
+            store_params['storage_options']['anon'] = dataset_config['Anonymous']
```
Comment (forman) [inline xcube/webapi/context.py:376]:
```suggestion
            storage_options['anon'] = dataset_config['Anonymous']
```

File: xcube/webapi/context.py [hunk 3]
```diff
@@ -352,28 +354,33 @@ def _maybe_assign_store_instance_ids(self):
...
         client_kwargs = dict(
         )
         if 'Endpoint' in dataset_config:
             client_kwargs['endpoint_url'] = dataset_config['Endpoint']
         if 'Region' in dataset_config:
             client_kwargs['region_name'] = dataset_config['Region']
-        store_params['client_kwargs'] = client_kwargs
+        store_params['storage_options']['client_kwargs'] = client_kwargs
```
Comment (forman) [inline xcube/webapi/context.py:383]:
```suggestion
        storage_options['client_kwargs'] = client_kwargs
        store_params = dict(storage_options=storage_options)
```

**Review by forman — COMMENTED** [review]
All fine, but I dislike subscript chainings and I dislike using string indexes, so I avoid them for clarity and simplicity. See suggestions.

**Pull request merged**